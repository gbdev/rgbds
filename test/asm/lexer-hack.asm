MACRO mac
	println "got {d:_NARG} args: \#"
ENDM

; indented, these were always macro invocations
	mac
	mac ro
	mac : ld a, 1

; in column 1, we historically treated these as labels
mac
mac ro
mac : ld b, 2

SECTION "test", ROM0

; a colon makes these into labels
	Label1: ld c, 3
Label2: ld d, 4

; a macro invocation when already defined as a label
	Label1 args
; and a label definition when already defined as a macro
mac: ld e, 5

; the space before the colon matters!
	undef :
undef :
