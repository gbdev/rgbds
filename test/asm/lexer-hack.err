error: lexer-hack.asm(22):
    "Label1" is not a macro
error: lexer-hack.asm(24):
    'mac' already defined at lexer-hack.asm(1)
error: lexer-hack.asm(27):
    Macro "undef" not defined
error: lexer-hack.asm(28):
    Macro "undef" not defined
error: Assembly aborted (4 errors)!
