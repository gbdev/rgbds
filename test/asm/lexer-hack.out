got 0 args: 
got 1 args: ro
got 2 args: : ld a,1
got 0 args: 
got 1 args: ro
got 2 args: : ld b,2
